packages/codegen-ui-golden-files/lib/react/forms/datastore-update-form-with-many-to-many-relationship/TagUpdateForm.jsx

```diff
@@ -277,38 +277,69 @@ export default function TagUpdateForm(props) {
           modelFields = onSubmit(modelFields);
         }
         try {
-          const postsToLink = [];
-          const postsToUnlink = [];
-          const postsSet = new Set();
-          const linkedPostsSet = new Set();
-          Posts.forEach((r) => postsSet.add(r.id));
-          linkedPosts.forEach((r) => linkedPostsSet.add(r.id));
+          const postsToLinkMap = new Map();
+          const postsToUnLinkMap = new Map();
+          const postsMap = new Map();
+          const linkedPostsMap = new Map();
+
+          Posts.forEach((r) => {
+            const count = postsMap.get(r.id);
+            const newCount = count ? count + 1 : 1;
```

Comment:
Why are we adding one to the count?

Reply:
According to the current UX, it allows adding duplicate records to the join table so have to track the count